### PR TITLE
added string tasks

### PR DESCRIPTION
**Pull request by nimdolphin**

### CODE REVIEW

File: string-tasks.js [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+// Функция должна быть нечувствительна к регистру:
+
+function checkSpam(str) {
+  let lowStr = str.toLowerCase();
+  let upperStr = str.toUpperCase();
+  if (lowStr.includes("viagra") || upperStr.includes("XXX")) {
+    return true;
+  } else {
+    return false;
+  }
```
Comment (kathrynowy) [inline string-tasks.js]: ?

File: string-tasks.js [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+// важность: 5
+// Напишите функцию ucFirst(str), возвращающую строку str с заглавным первым символом.
+
+function ucFirst(str) {
+  if (!str) {
+    return str;
+  } else {
+    return str[0].toUpperCase() + str.slice(1);
+  }
+}
```
Comment (kathrynowy) [inline string-tasks.js]: write in 1 line with an arrow func and implicit return

File: string-tasks.js [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+// // Результатом функции должна быть та же строка,
+//  если усечение не требуется, либо, если необходимо,
+//   усечённая строка.
+
+function truncate(str, maxlength) {
+  if (str.length > maxlength) {
+    return str.slice(0, maxlength - 1) + "…";
+  } else {
+    return str;
+  }
```
Comment (kathrynowy) [inline string-tasks.js]: same here

File: string-tasks.js [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+console.log(truncate("Всем привет!", 20));
+
+// Выделить число
+// Есть стоимость в виде строки "$120". То есть сначала
+// идёт знак валюты, а затем – число.
+// // Создайте функцию extractCurrencyValue(str), которая
+// будет из такой строки выделять числовое значение и возвращать его.
+
+function extractCurrencyValue(str) {
+  return +str.slice(1);
```
Comment (kathrynowy) [inline string-tasks.js]: what if str will come 'dfdf'

try it out

File: string-tasks.js [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+}
+
+logAllChars("hello");
+
+// 1. Дана строка 'frontent$developer$wow. Замените все $ на *
+// с помощью глобального поиска и замены.
+
+const str1 = "frontent$developer$wow";
+
+const newStr1 = str1.replace(/[$]/gi, "*");
```
Comment (kathrynowy) [inline string-tasks.js]: what does it mean gi?

Comment (kathrynowy) [reply string-tasks.js]: don't we have replaceAll?

File: string-tasks.js [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+console.log(newStr1);
+
+// 2.  Напишите функцию convertStringToArray(str), которая преобразует строку в массив слов.
+
+const str2 = "I am learning frontend development";
+
+const convertStringToArray = (str2) => {
+  return str2.trim().split(" ");
+};
```
Comment (kathrynowy) [inline string-tasks.js]: why do u use explicit return?

File: string-tasks.js [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+
+console.log(result); // ['I', 'am', 'learning', 'frontend', 'development']
+
+// 3. Напишите функцию removeSymbols(str, length), которая
+// возвращает подстроку, состоящую из указанного количества символов.
+
+const str3 = "hey, buddy";
+
+function removeSymbols(str3, length) {
+  if (str3 === String || length > 0) {
```
Comment (kathrynowy) [inline string-tasks.js]: ??

1 how can "hey hubby" be an object like String?

image


2 we don;t check >0 like this if there is only range (0 - ..... any amount)

you can check for zero !length
and for other values - just  (length)


I described it in TG many times, please follow it

File: string-tasks.js [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+console.log(removeSymbols(str3, 4));
+
+// 4. Напишите функцию pasteDashesIntoString(str), которая принимает
+// строку str в качестве аргумента и вставляет тире (-) между словами.
+// При этом все символы строки необходимо перевести в верхний регистр.
+
+const str4 = "I love pizza";
+function pasteDashesIntoString(str4) {
+  return str4.toUpperCase().split(" ").join("-");
+}
```
Comment (kathrynowy) [inline string-tasks.js]: arrow func

File: string-tasks.js [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+console.log(pasteDashesIntoString(str4));
+
+// 5. Напишите функцию makeStringLowerCased(str), которая
+// принимает строку str в качестве аргумента и переводит в нижний регистр.
+
+function makeStringLowerCased(str5) {
+  let newStr = "";
+  newStr = str5.toLowerCase();
+  console.log(newStr);
+  return newStr;
```
Comment (kathrynowy) [inline string-tasks.js]: do u need a new variable/clg?

please make your code clean

File: string-tasks.js [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+
+// 6.Напишите функцию capitaliseAllWords(str), которая возвращает
+// строку, в которой каждое слово начинается с заглавной буквы.
+
+function capitaliseAllWords(str) {
+  let newArr = str
+    .split(" ")
+    .map((item) => item[0].toUpperCase() + item.slice(1))
+    .join(" ");
+  console.log(newArr);
```
Comment (kathrynowy) [inline string-tasks.js]: your function is returning nothing

it's useless then

File: string-tasks.js [hunk 11]
```diff
@@ -0,0 +1,339 @@
...
+}
+
+capitaliseAllWords("i love pizza");
+
+// 7. Напишите функцию makeFirstLetterUpperCased(str), которая
+// принимает строку str в качестве аргумента и делает первую букву
+// заглавной.
+
+function makeFirstLetterUpperCased(str) {
+  console.log(str[0].toUpperCase() + str.slice(1));
```
Comment (kathrynowy) [inline string-tasks.js]: same

File: string-tasks.js [hunk 12]
```diff
@@ -0,0 +1,339 @@
...
+    let newStr = str7[i];
+    let bigW = newStr.toUpperCase();
+    let smallW = newStr.toLowerCase();
+
+    if (newStr === newStr.toLowerCase()) {
+      // console.log(bigW);
+      newS += bigW;
+    } else {
+      // console.log(smallW);
+      newS += smallW;
```
Comment (kathrynowy) [inline string-tasks.js]: can u do it in 1 line?

Comment (kathrynowy) [reply string-tasks.js]: and do u really need bigW variable?

and what does it mean W??

File: string-tasks.js [hunk 13]
```diff
@@ -0,0 +1,339 @@
...
+const str5 = prompt("джаваскрипт", "");
+
+function changeLetterRegister(str5) {
+  let strNew5 = "";
+
+  for (i = 0; i < str5.length; i++) {
+    // console.log(str5[i]);
+
+    if (!(i % 2)) {
+      console.log(str5[i].toLowerCase());
```
Comment (kathrynowy) [inline string-tasks.js]: ?? remove your logs when you push the code into repository

File: string-tasks.js [hunk 14]
```diff
@@ -0,0 +1,339 @@
...
+      strNew5 += str5[i].toLowerCase();
+    } else if (i % 2 === 1) {
+      console.log(str5[i].toUpperCase());
+      strNew5 += str5[i].toUpperCase();
```
Comment (kathrynowy) [inline string-tasks.js]: same in 1 line

File: string-tasks.js [hunk 15]
```diff
@@ -0,0 +1,339 @@
...
+  let c = "+" === sign;
+  let b = "-" === sign;
+
+  let newNum = "";
+
+  if (a && c) {
+    newNum = +(num + "0".repeat(len - num.length));
+  } else if (a && b) {
+    newNum = -(num + "0".repeat(len - num.length));
+  }
```
Comment (kathrynowy) [inline string-tasks.js]: can you just return make 1 result?


create a string with zeros
and add sign there

then convert to number.

File: string-tasks.js [hunk 16]
```diff
@@ -0,0 +1,339 @@
...
+  console.log(newStr.indexOf(char));
+
+  let d = "after" === pos;
+  let f = "before" === pos;
+
+  if (newStr.includes(char) && d) {
+    newStr2 = newStr.slice(0, 1).join(" ");
+  } else if (newStr.includes(char) && f) {
+    newStr2 = newStr.slice(0, 3).join(" ");
+  }
```
Comment (kathrynowy) [inline string-tasks.js]: can you do it in 1 line?
just slice a needed amount

**Review by kathrynowy — CHANGES_REQUESTED** [review]

File: string-tasks.js (hunk 1 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 2 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 3 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 4 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 5 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 6 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 7 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 8 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 9 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 10 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 11 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 12 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 13 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 14 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 15 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 16 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js [hunk 17]
```diff
@@ -0,0 +1,252 @@
...
+
+const str = "Javascript — синхронный язык программирования";
+
+function getSubstring(str, char, pos) {
+  let newStr = str.split(" ");
+  let idx = newStr.indexOf(char);
+
+  let newString = pos
+    ? newStr.slice(0, idx).join(" ")
+    : newStr.slice(0, idx).join(" ");
```
Comment (kathrynowy) [inline string-tasks.js]: ?

File: string-tasks.js [hunk 18]
```diff
@@ -0,0 +1,252 @@
...
+// то на выходе должно быть «дЖаВаСкРиПт».
+
+const str5 = prompt("джаваскрипт", "");
+
+function changeLetterRegister(str5) {
+  let strNew5 = "";
+
+  for (i = 0; i < str5.length; i++) {
+    let strN = !(i % 2) ? str5[i].toLowerCase() : str5[i].toUpperCase();
+    strNew5 += strN;
```
Comment (kathrynowy) [inline string-tasks.js]:
```suggestion
    let strN = !(i % 2) ? str5[i].toLowerCase() : str5[i].toUpperCase();
    
    strNew5 += strN;
```

File: string-tasks.js [hunk 19]
```diff
@@ -0,0 +1,252 @@
...
+  }
+  return strNew5;
```
Comment (kathrynowy) [inline string-tasks.js]:
```suggestion
  }
  
  return strNew5;
```

File: string-tasks.js [hunk 20]
```diff
@@ -0,0 +1,252 @@
...
+
+function changeLetter(str7) {
+  let newS = "";
+  for (i = 0; i < str7.length; i++) {
+    let newStr = str7[i];
+    let bigLetter = newStr.toUpperCase();
+    let smallLetter = newStr.toLowerCase();
+
+    let newS2 = newS === smallLetter ? bigLetter : smallLetter;
+
```
Comment (kathrynowy) [inline string-tasks.js]:
```suggestion

    let newS2 = smallLetter;
    
    if (newS === smallLetter){
    newS2 = bigLetter ;
    }

```

File: string-tasks.js [hunk 21]
```diff
@@ -0,0 +1,252 @@
...
+const pasteDashesIntoString = (str4) => str4.toUpperCase().split(" ").join("-");
+
+pasteDashesIntoString(str4);
+
+// 5. Напишите функцию makeStringLowerCased(str), которая
+// принимает строку str в качестве аргумента и переводит в нижний регистр.
+
+function makeStringLowerCased(str5) {
+  let newStr = str5.toLowerCase();
+  return newStr;
```
Comment (kathrynowy) [inline string-tasks.js]: 1 line all func

**Review by kathrynowy — CHANGES_REQUESTED** [review]

File: string-tasks.js (hunk 17 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 18 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 19 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 20 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 21 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js [hunk 22]
```diff
@@ -0,0 +1,253 @@
...
+const result = convertStringToArray(str2);
+
+console.log(result); // ['I', 'am', 'learning', 'frontend', 'development']
+
+// 3. Напишите функцию removeSymbols(str, length), которая
+// возвращает подстроку, состоящую из указанного количества символов.
+
+const str3 = "hey, buddy";
+
+const removeSymbols = (str3, length) =>
```
Comment (kathrynowy) [inline string-tasks.js]: Точно ли такое словио должно быть? Если нет строки или нет длины - ты обрезаешь
иначе нет

должно быть наоборот?

File: string-tasks.js [hunk 23]
```diff
@@ -0,0 +1,253 @@
...
+  return a === b;
+}
+
+compareStrings("string", "StRiNg");
+compareStrings("JavaScript", "JavaScriptonite");
+
+// 12. Напишите функцию repeatStr(str, n), которая возвращает строку
+//  повторяемую определённое количество раз.
+
+function repeatStr(str, n) {
```
Comment (kathrynowy) [inline string-tasks.js]: In1 line

**Review by kathrynowy — APPROVED** [review]

File: string-tasks.js (hunk 22 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

File: string-tasks.js (hunk 23 above)
Comment (nimdolphin) [reply string-tasks.js]: fixed

**Pull request merged**